From 996154202cd9d6f751f5046a7bd7ce53182c01e4 Mon Sep 17 00:00:00 2001
From: Paulo Zanoni <paulo.r.zanoni@intel.com>
Date: Tue, 19 Jan 2016 11:35:55 -0200
Subject: [PATCH 0926/2508] drm/i915/fbc: don't store the fb_id on reg_params

commit b20d27526cb0e8dd2b7db3b45ec96aecfc02d610 upstream

We don't actually use fb_id anywhere. We already compare all
parameters that matter to the hardware: pixel format, stride,
fence_reg and ggtt_offset. The ID shouldn't make a difference.

Besides, we already update the FBC data at every modeset/flip, so this
can't change behind our backs.

Reviewed-by: Maarten Lankhorst <maarten.lankhorst@linux.intel.com>
Signed-off-by: Paulo Zanoni <paulo.r.zanoni@intel.com>
Link: http://patchwork.freedesktop.org/patch/msgid/1453210558-7875-23-git-send-email-paulo.r.zanoni@intel.com
---
 drivers/gpu/drm/i915/i915_drv.h  |    2 --
 drivers/gpu/drm/i915/intel_fbc.c |    2 --
 2 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 88863b9..25c55d7 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -934,7 +934,6 @@ struct intel_fbc {
 
 		struct {
 			u64 ilk_ggtt_offset;
-			uint32_t id;
 			uint32_t pixel_format;
 			unsigned int stride;
 			int fence_reg;
@@ -951,7 +950,6 @@ struct intel_fbc {
 
 		struct {
 			u64 ggtt_offset;
-			uint32_t id;
 			uint32_t pixel_format;
 			unsigned int stride;
 			int fence_reg;
diff --git a/drivers/gpu/drm/i915/intel_fbc.c b/drivers/gpu/drm/i915/intel_fbc.c
index cdd99cf..35e92bc 100644
--- a/drivers/gpu/drm/i915/intel_fbc.c
+++ b/drivers/gpu/drm/i915/intel_fbc.c
@@ -735,7 +735,6 @@ static void intel_fbc_update_state_cache(struct intel_crtc *crtc)
 	 * platforms that need. */
 	if (dev_priv->fbc.activate == ilk_fbc_activate)
 		cache->fb.ilk_ggtt_offset = i915_gem_obj_ggtt_offset(obj);
-	cache->fb.id = fb->base.id;
 	cache->fb.pixel_format = fb->pixel_format;
 	cache->fb.stride = fb->pitches[0];
 	cache->fb.fence_reg = obj->fence_reg;
@@ -858,7 +857,6 @@ static void intel_fbc_get_reg_params(struct intel_crtc *crtc,
 	params->crtc.plane = crtc->plane;
 	params->crtc.fence_y_offset = get_crtc_fence_y_offset(crtc);
 
-	params->fb.id = cache->fb.id;
 	params->fb.pixel_format = cache->fb.pixel_format;
 	params->fb.stride = cache->fb.stride;
 	params->fb.fence_reg = cache->fb.fence_reg;
-- 
1.7.5.4

